app/queries/my_facilities/missed_visits_query.rb

```diff
@@ -0,0 +1,116 @@
+class MyFacilities::MissedVisitsQuery
+  # Wrap query method calls with the appropriate timezone in which the reports will be consumed
+  # This is probably the ENV['ANALYTICS_TIME_ZONE']
+  # Example: `Time.use_zone('timezone string') { bp_control_query_object.cohort_registrations }`
+
+  include QuarterHelper
+  include MonthHelper
+
+  REGISTRATION_BUFFER = 2.months
+
+  attr_reader :periods
+
+  def initialize(facilities: Facility.all, period: :quarter, last_n: 3)
+    # period can be :quarter, :month.
+    # last_n is the number of quarters/months data to be returned
+    @facilities = facilities
+    @period = period
+    @periods = period_list(period, last_n)
+    @latest_period = @periods.first
+  end
+
+  def patients
+    @patients ||=
+      Patient
+      .joins('INNER JOIN latest_blood_pressures_per_patients
+              ON patients.id = latest_blood_pressures_per_patients.patient_id')
+      .where(latest_blood_pressures_per_patients: { bp_facility_id: @facilities })
+      .where('recorded_at < ?', Time.current.beginning_of_day - REGISTRATION_BUFFER)
+  end
+
+  def patients_by_period
+    @patients_by_period ||=
+      @periods.map do |year, period|
+        period_start = (@period == :quarter ? local_quarter_start(year, period) : local_month_start(year, period))
+        [[year, period], patients.where('patient_recorded_at < ?', period_start - REGISTRATION_BUFFER)]
```

Comment:
Is this assembling patients from, say, Jan-Mar, Apr-Jun, Jul-Sept, or is it doing (beginning of time - Jan), (beginning of time - Apr), (beginning of time - Sept)?

Reply:
The second -> Beginning of time up to the point the period starts

Reply:
Cool. Wanted to confirm that that's the intention.